dds_web/api/project.py

```diff
@@ -188,6 +188,107 @@ def post(self):
 
         return {"message": return_message}
 
+    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
+    @logging_bind_request
+    @json_required
+    @handle_validation_errors
+    def patch(self):
+        """Partially update a the project status"""
+        # Get project ID, project and verify access
+        project_id = dds_web.utils.get_required_item(obj=flask.request.args, req="project")
+        project = dds_web.utils.collect_project(project_id=project_id)
+        dds_web.utils.verify_project_access(project=project)
+
+        # Get json input from request
+        json_input = flask.request.get_json(silent=True)  # Already checked by json_required
+
+        # Operation must be confirmed by the user - False by default
+        confirmed_operation = json_input.get("confirmed", False)
+        if not isinstance(confirmed_operation, bool):
+            raise DDSArgumentError(message="`confirmed` is a boolean value: True or False.")
+        if not confirmed_operation:
+            warning_message = "Operation must be confirmed before proceding."
+            return {"warning": warning_message}
+
+        # Cannot change project status if project is busy
+        if project.busy:
+            raise ProjectBusyError(
+                message=(
+                    f"The deadline for the project '{project_id}' is already in the process of being changed. "
+                    "Please try again later. \n\nIf you know the project is not busy, contact support."
+                )
+            )
+
+        self.set_busy(project=project, busy=True)
+
+        # Extend deadline
+        try:
+            new_deadline_in = json_input.get(
+                "new_deadline_in", None
+            )  # if not provided is None -> deadlie is not updated
```

Comment:
This also means you can remove the `if new_deadline_in` and `else: return_message="Nothing to update"`

Reply:
So the idea is that the unit user shouldn't need to specify the `deadline` in the CLI command, and that if they don't specify it it goes to the default -- `Unit.days_in_available`. But it would be good to display that to the users when checking for confirmation. "You're attempting to extend the deadline with x days". The problem is atm though that `ProjectStatus.get` doesn't return that.

Reply:
So it seems that either we 
1. Keep it like it is now: `ProjectStatus.get` first and then `ProjectStatus.patch` to do the deadline extension, and in this case we force the user to specify the `deadline` in the CLI. 
2. Change the `ProjectStatus.patch` endpoint to check the project status, return the project status + what they're attempting to extend the deadline with, and then the CLI calls `ProjectStatus.patch` a second time which (if `confirmed=True` extends the deadline

Reply:
But if in the future the function is extended to update only other parameters, then it will always fetch a new_deadline attribute, making it difficult to filter out.

Also, the default _days_in_available_ is 90. And the function is checking for _new_deadline_in + current_deadline > 90:_
This check will always fail when using new_deadline = 90

Reply:
Let me know if I'm not explaining it clearly etc

Reply:
Yeah, the comment was written before refreshing to see the full feedback. But my point stands with the option 1. Because of what I mentioned. 

If we don't want to "force" them to write it in the command. Then it can be asked as a terminal input when project status extend is invoked

Reply:
Discussed via zoom. Agreed upon: 

```python
        if not confirmed_operation:
            warning_message = "Operation must be confirmed before proceding."
            return {"warning": warning_message}
```

Should return 
1. current project status
2. The default number of days_in_available (project.responsible_unit.days_in_available) 
3. Possibly the message